Events/CheckedRateLimitEvent.php

```diff
@@ -3,49 +3,97 @@
 namespace Noxlogic\RateLimitBundle\Events;
 
 use Noxlogic\RateLimitBundle\Annotation\RateLimit;
-use Symfony\Component\EventDispatcher\Event;
+use Symfony\Component\HttpKernel\Kernel;
+use Symfony\Contracts\EventDispatcher\Event as ContractsEvent;
+use Symfony\Component\EventDispatcher\Event as ComponentEvent;
 use Symfony\Component\HttpFoundation\Request;
 
-class CheckedRateLimitEvent extends Event
-{
+if(Kernel::VERSION_ID >= 40300) {
```

Comment:
As a suggestion, you could put the two versions into two different and properly named files, and then in the actual `CheckedRateLimitEvent.php` you'd put this conditional, `require`-ing the correct file in each branch.